neptune/new/attributes/namespace.py

```diff
@@ -75,8 +74,14 @@ def _collect_atom_values(self, attribute_dict) -> dict:
         return result
 
     def fetch(self) -> dict:
-        namespace_values = self._backend.fetch_atom_attribute_values(self._run_uuid, self._path)
-        return self._collect_atom_values(namespace_values)
+        attributes = self._backend.fetch_atom_attribute_values(self._run_uuid, self._path)
+        from neptune.new.internal.run_structure import RunStructure
+        from neptune.new.internal.utils.paths import parse_path
+        run_struct = RunStructure()
+        prefix_len = len(self._path)
+        for attr_name, attr_type, attr_value in attributes:
+            run_struct.set(parse_path(attr_name)[prefix_len:], (attr_type, attr_value))
```

Comment:
I think this "deserialization" logic should be placed in the RunStructure class.

Reply:
do you mean `parse_path(attr_name)[prefix_len:]`?

the thing is I'm abusing `RunStructure` a little bit here (it's normally used for run **structure**, not values) so probably the by-the-book thing to do would to subclass `RunStructure` as `RunWithValuesStructure` and add deserialization there... not sure if it's worth it, though?

Reply:
So maybe in some utils file/internal method in Namespace?